build.gradle

```diff
@@ -49,6 +49,8 @@ dependencies {
     testRuntimeOnly group: 'org.junit.jupiter', name: 'junit-jupiter-engine', version: jupiterVersion
     testImplementation group: 'org.mockito', name: 'mockito-core', version: '3.3.3'
     testImplementation group: 'com.github.stefanbirkner', name: 'system-rules', version: '1.19.0'
+    testImplementation 'junit:junit:4.11'
```

Comment:
From `gradleFixLint` task

Reply:
According to `gradle dependecies` this comes from the `com.github.stefanbirkner` dependency which is not used any more in our code actively.  We also had some `Assert.asserTrue()` commands which were from the Junit4 library.  I have created a new PR to your branch as a suggestion. Feel free to make any changes you wish, but the stefanbirkner, junit4 and hamcrest1.3 dependencies are not necessary 

In general when we see  junit4 dependencies, we should remove them like weeds from crops, because they do not necessarily execute and we get a false image that our tests succeed

Reply:
I merged your branch into this.